### ISSUE TITLE
`#pragma warning disable` compiler directives are not working for hotspot rules

### ISSUE THREAD

**Issue by costin-zaharia-sonarsource** [SonarSource/sonar-dotnet#4724]
### Description

`#pragma warning disable` compiler directives are not working for hotspot rules. These work just fine for non-hotspot rules. We need to check if this is a problem on our side or on the Roslyn compiler.

### Repro steps

Create a project with a file containing:
```
#pragma warning disable S4790

using System;
using System.Security.Cryptography;

Console.WriteLine("Test");

var algorithm = HashAlgorithm.Create("SHA1") ?? throw new ArgumentException("Failed to create Sha1 hash algorithm");
```

### Expected behavior

Warning `S4790` should not be raised since it's disabled using the [#pragma directive](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/preprocessor-directives#pragma-warning)

### Actual behavior

The warning is still raised.

### Known workarounds

Use the SonarQube configuration.

### Related information

* C#/VB.NET Plugins version: 8.26
* Microsoft (R) Build Engine version 16.10.2+857e5a733 for .NET

**Comment by pavel-mikula-sonarsource** [comment]
This is likely related to the fact, that hotspot rules have `NotConfigurable` tag on them. As written in the issue, we need to understand what the `#pragma` is supposed to do with these on Roslyn side.

https://github.com/SonarSource/sonar-dotnet/blob/b2dbe9bdb4b5f34542986d97190b1ab7ec4f262d/analyzers/src/SonarAnalyzer.Common/DiagnosticAnalyzer/DiagnosticDescriptorBuilder.cs#L75

**Comment by pavel-mikula-sonarsource** [comment]
After discussion with @tom-howlett-sonarsource, hotspots should be configurable. 
We need to delete this line
https://github.com/SonarSource/sonar-dotnet/blob/cf620cff6e21ee8990285f5fcda92c6cd4cb7cd3/analyzers/src/SonarAnalyzer.Common/DiagnosticAnalyzer/DiagnosticDescriptorFactory.cs#L62

and fix all UTs that will fail

**Comment by fauresco** [comment]
I can still see this behavior in SonarQube running in AzureDevops build pipeline. Despite having:

```
# pragma warning disable S5122
<code>
# pragma warning restore S5122
```

I still see the warnings in the build output.

Maybe the version of the csharp plugin loaded by the AzurePipelines does not have this fix yet? 

Here is the log files of the build pipeline:

Task         : Prepare Analysis Configuration
Description  : Prepare SonarQube analysis configuration
Version      : 5.18.4
Author       : sonarsource
Help         : Version: 5.18.4.

SonarScanner for MSBuild 5.15
Using the .NET Core version of the Scanner for MSBuild
Pre-processing started.
Preparing working directories...
19:15:25.679  Updating build integration targets...
19:15:26.275  Fetching analysis configuration settings...
19:15:26.515  Provisioning analyzer assemblies for cs...
19:15:26.516  Installing required Roslyn analyzers...
19:15:26.517  Processing plugin: csharp version 8.36.1.44192
19:15:26.656  Processing plugin: vbnet version 8.36.1.44192
19:15:26.78  Provisioning analyzer assemblies for vbnet...
19:15:26.78  Installing required Roslyn analyzers...
19:15:26.78  Processing plugin: csharp version 8.36.1.44192
19:15:26.781  Processing plugin: vbnet version 8.36.1.44192
19:15:26.788  Incremental PR analysis is available starting with SonarQube 9.9 or later.
19:15:26.788  Cache data is empty. A full analysis will be performed.
19:15:26.827  Pre-processing succeeded.
Finishing: Sonarqube Prepare Analysis Configuration

**Comment by mary-georgiou-sonarsource** [comment]
@fauresco this is correct!
The fix is included in [version 8.43](https://github.com/SonarSource/sonar-dotnet/releases/tag/8.43.0.51858) and from the logs I see that `8.36 is used`.  See line: `19:15:26.517 Processing plugin: csharp version 8.36.1.44192`

**Comment by fauresco** [comment]
> @fauresco this is correct! The fix is included in [version 8.43](https://github.com/SonarSource/sonar-dotnet/releases/tag/8.43.0.51858) and from the logs I see that `8.36 is used`. See line: `19:15:26.517 Processing plugin: csharp version 8.36.1.44192`

@mary-georgiou-sonarsource do you know how can I use the version 8.43+ in Azure DevOps?

In my build pipeline yaml file I have the sonarqube tasks like:

- task: SonarQubePrepare@5
- task: SonarQubeAnalyze@5
- task: SonarQubePublish@5

Or maybe this is something I have to wait for the "@6" version of the sonar tasks?
